src/ui/components/Navigation/Base/configure.tsx

```diff
@@ -0,0 +1,293 @@
+            place: PLACE.DATASETS,
+            submenu: 'other',
+        },
+    ];
+});
+
+export const getCreatableEntries = memoize(
+    ({
+        onClick,
+        place,
+        isOnlyCollectionsMode,
+        creatableEntriesConfig,
+        b,
+    }: CreateEntryProps & {
+        b: (title: string) => string;
+        creatableEntriesConfig: EntrySettings[];
+    }) => {
+        const menuItems: DropdownMenuItemMixed<() => void>[] = [];
+        const menuChartItems: DropdownMenuItem<() => void>[] = [];
+        const menuOtherItems: DropdownMenuItem<() => void>[] = [];
+
+        creatableEntriesConfig.forEach((entryConfig) => {
+            if (entryConfig.condition && entryConfig.condition() === false) {
+                return;
+            }
+
+            let targetMenu;
+            if (entryConfig.submenu === 'charts') {
+                targetMenu = menuChartItems;
+            } else if (entryConfig.submenu === 'other') {
+                targetMenu = menuOtherItems;
+            } else {
+                targetMenu = menuItems;
+            }
+
+            targetMenu.push({
+                action: () => onClick(entryConfig.value),
+                icon: entryConfig.icon,
+                text: <Title className={b('item-title')} title={entryConfig.text} />,
+            });
+        });
+
+        // If current menu contains Charts only then return Charts creation subset
+        if (place === PLACE.WIDGETS) {
+            return menuChartItems;
+        }
+
+        if (isOnlyCollectionsMode === false) {
+            menuItems.push({
+                action: () => onClick(CreateMenuValue.Folder),
+                icon: <EntityIcon type="folder" iconSize={18} />,
+                text: <Title className={b('item-title')} title={i18n('value_create-folder')} />,
+            });
+        }
+
+        menuItems.push(menuChartItems, menuOtherItems);
+
+        return menuItems;
+    },
+);
+
+export const getCreateEntrySwitcher = memoize(
+    ({
+        place,
+        onClick,
+        withMenu,
+        placesConfig,
+    }: {
+        place: string;
+        onClick: (value: CreateMenuValue, options?: Record<string, unknown>) => void;
+        withMenu: boolean;
+        placesConfig: PlaceParameterItem[];
+    }) => {
+        const targetPlace = placesConfig.find((placeConfig) => {
+            return placeConfig.place === place;
+        });
+
+        if (!targetPlace) {
+            return null;
+        }
+
+        return (
+            <Button
+                view="action"
+                qa="create-entry-button"
+                className={b('button-create')}
+                onClick={
+                    withMenu
+                        ? undefined
+                        : () => {
+                              onClick(targetPlace.value);
+                          }
+                }
+            >
+                {targetPlace.buttonText}
+            </Button>
+        );
+    },
```

Comment:
It would be good to split this file into two parts - react components and utilities.
And it's a bit strange to see the components in the `configure` file